pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -186,7 +186,7 @@ public CompletableFuture<Boolean> canProduceAsync(TopicName topicName, String ro
             if (isSuperUser) {
                 return CompletableFuture.completedFuture(true);
             } else {
-                return provider.canProduceAsync(topicName, role, authenticationData);
+                return provider.allowTopicOperationAsync(topicName, role, TopicOperation.PRODUCE, authenticationData);
```

Comment:
Please remove the superuser check, because the `provider.allowTopicOperationAsync` includes that.